### ISSUE TITLE
1.20.0 rc1 & 1.20.0 rc3 Crash under same circumstance

### ISSUE THREAD

**Issue by MtSheepFarm** [anegostudios/VintageStory-Issues#4692]
### Game Version

1.20.0 rc3

### Platform

Windows

### Modded

Modded

### SP/MP

Singleplayer

### Description

Game completely crashes upon death both yesterday in v 1.20.0 rc1 and today in v 1.20.0 rc3 It may be important to note that I was returning to a death point (friggin bears man) and so the "death crash" comes within minutes of a successful respawn. Yesterday when it crashed, I thought it was perhaps just because I had been playing for a while and I did not reboot the game afterwards. Today I rebooted it after updating the game, spawning in at what I assumed was the last auto-save point (about 2 minutes before death and crash) and when I again died to the bear, it immediately crashed before respawn. As this happened twice at the same point, I thought it significant enough to report. 

### How to reproduce

Die 2-3 times ~1k blocks away from respawn point? I'm honestly not sure how to recreate the issue. Crash log is included. 

### Logs

```
Running on 64 bit Windows 10.0.19045.0 with 32649 MB RAM
Game Version: v1.20.0-rc.3 (Unstable)
12/14/2024 11:34:57 AM: Critical error occurred
Loaded Mods: bettersticks@1.0.0, landformoverhaul@0.0.2, OBS@0.0.1, someextramolds@1.0.6, game@1.20.0-rc.3, coolinbarrel@0.5.3, freedomunits@1.1.2, creative@1.20.0-rc.3, survival@1.20.0-rc.3
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.Client.NoObf.ClientChunkData.EmptyAndReuseArrays(List`1 datas) in VintagestoryLib\Client\Model\ClientChunkData.cs:line 197
   at Vintagestory.Common.ChunkDataPool.FreeArraysAndReset(ChunkData cdata) in VintagestoryLib\Common\Datastructures\ChunkDataPool.cs:line 89
   at Vintagestory.Common.WorldChunk.Dispose() in VintagestoryLib\Common\Datastructures\WorldChunk.cs:line 461
   at Vintagestory.Client.NoObf.SystemUnloadChunks.HandleChunkUnload(Packet_Server packet) in VintagestoryLib\Client\Systems\UnloadChunks.cs:line 87
   at Vintagestory.Client.NoObf.ClientMain.ExecuteMainThreadTasks(Single deltaTime) in VintagestoryLib\Client\ClientMain.cs:line 785
   at Vintagestory.Client.GuiScreenRunningGame.RenderToPrimary(Single dt) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 161
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 695
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 670
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 92
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 334
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass10_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 130
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93

Event Log entries for Vintagestory.exe, the latest 3
==================================
{ TimeGenerated = 12/7/2024 6:05:28 PM, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.20.0.0, time stamp: 0x66470000
Faulting module name: coreclr.dll, version: 7.0.1023.36312, time stamp: 0x64b06d6c
Exception code: 0xc0000005
Fault offset: 0x00000000001c98a6
Faulting process id: 0x1468
Faulting application start time: 0x01db490d39eb7046
Faulting application path: C:\Users\Noah\AppData\Roaming\Vintagestory\Vintagestory.exe
Faulting module path: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\7.0.10\coreclr.dll
Report Id: 7509d8cb-1daa-4fec-a724-a110ea49daf1
Faulting package full name: 
Faulting package-relative application ID:  }
--------------
{ TimeGenerated = 12/6/2024 10:07:46 AM, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.20.0.0, time stamp: 0x66470000
Faulting module name: coreclr.dll, version: 7.0.1023.36312, time stamp: 0x64b06d6c
Exception code: 0xc0000005
Fault offset: 0x00000000001c98a6
Faulting process id: 0x3830
Faulting application start time: 0x01db48014c97f6b5
Faulting application path: C:\Users\Noah\AppData\Roaming\Vintagestory\Vintagestory.exe
Faulting module path: C:\Program Files\dotnet\shared\Microsoft.NETCore.App\7.0.10\coreclr.dll
Report Id: f72f2cae-c246-4ac6-a1ff-6aceb4b83111
Faulting package full name: 
Faulting package-relative application ID:  }
--------------
{ TimeGenerated = 12/5/2024 11:06:18 AM, Site = , Source = Application Error, Message = Faulting application name: Vintagestory.exe, version: 1.20.0.0, time stamp: 0x66470000
Faulting module name: openal32.dll, version: 1.23.0.0, time stamp: 0x63dd31ad
Exception code: 0x40000015
Fault offset: 0x00000000000df046
Faulting process id: 0x26a4
Faulting application start time: 0x01db473f85ae7e25
Faulting application path: C:\Users\Noah\AppData\Roaming\Vintagestory\Vintagestory.exe
Faulting module path: C:\Users\Noah\AppData\Roaming\Vintagestory\Lib\openal32.dll
Report Id: 92cd3917-75ba-4cec-b373-5feec84fcca1
Faulting package full name: 
Faulting package-relative application ID:  }
```

**Comment by MtSheepFarm** [comment]
One other thing that just came to mind, I was respawning on a Temporal Gear set point. I have the respawns set to infinite because I die a lot. I know nothing about how any of that works "under the hood" but I thought it might be important to note just in case its a bug with that system and not just a loading error.

**Comment by radfast** [comment]
Thank you.  If there is any chance of you still having the earlier crash (the first one from a few minutes before this one, I think you said?) it would be very helpful to see that, to be sure they are indeed the same cause.   Crashes can be collected together in a single file `client-crash.log` or you can check individual older `client-main.log` files in the `Archive` subfolder of your `Logs` folder.

Or post a ZIP of your entire `Logs` folder if you prefer.

The crash posted in the first post in this thread, appears to be an extremely rare deep engine crash - nothing specifically relating to this aggressive bear or anything you did as a player.  Hopefully you have not experienced many crashes like this on other occasions when you have re-spawned?

Provisionally, I believe the root cause of this is fixed in next upcoming version 1.20-rc.5, but to make sure that I'm on the right track it would be great to see that earlier log as requested above.

**Comment by MtSheepFarm** [comment]
@radfast I believe this is the log from that first crash that I had mentioned. Incidentally, after I made the bug report, I had the game crash once more on death, but with no log generated, and the game loading me in at my appropriate spawn point upon reboot. No further crashing so far. Thank you for your help and your time looking into this. Please let me know if there is any other helpful information I might be able to provide. 

Game Version: v1.20.0-rc.1 (Unstable)
12/13/2024 1:28:47 PM: Critical error occurred
Loaded Mods: bettersticks@1.0.0, landformoverhaul@0.0.2, OBS@0.0.1, someextramolds@1.0.6, game@1.20.0-rc.1, coolinbarrel@0.5.3, freedomunits@1.1.2, creative@1.20.0-rc.1, survival@1.20.0-rc.1
System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.Client.NoObf.ClientChunkData.EmptyAndReuseArrays(List`1 datas) in VintagestoryLib\Client\Model\ClientChunkData.cs:line 197
   at Vintagestory.Common.ChunkDataPool.FreeArraysAndReset(ChunkData cdata) in VintagestoryLib\Common\Datastructures\ChunkDataPool.cs:line 89
   at Vintagestory.Common.WorldChunk.Dispose() in VintagestoryLib\Common\Datastructures\WorldChunk.cs:line 455
   at Vintagestory.Client.NoObf.SystemUnloadChunks.HandleChunkUnload(Packet_Server packet) in VintagestoryLib\Client\Systems\UnloadChunks.cs:line 87
   at Vintagestory.Client.NoObf.ClientMain.ExecuteMainThreadTasks(Single deltaTime) in VintagestoryLib\Client\ClientMain.cs:line 785
   at Vintagestory.Client.GuiScreenRunningGame.RenderToPrimary(Single dt) in VintagestoryLib\Client\MainMenu\Screens\GuiScreenRunningGame.cs:line 161
   at Vintagestory.Client.ScreenManager.Render(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 695
   at Vintagestory.Client.ScreenManager.OnNewFrame(Single dt) in VintagestoryLib\Client\ScreenManager.cs:line 670
   at Vintagestory.Client.NoObf.ClientPlatformWindows.window_RenderFrame(FrameEventArgs e) in VintagestoryLib\Client\ClientPlatform\GameWindow.cs:line 92
   at OpenTK.Windowing.Desktop.GameWindow.Run()
   at Vintagestory.Client.ClientProgram.Start(ClientProgramArgs args, String[] rawArgs) in VintagestoryLib\Client\ClientProgram.cs:line 329
   at Vintagestory.Client.ClientProgram.<>c__DisplayClass10_0.<.ctor>b__1() in VintagestoryLib\Client\ClientProgram.cs:line 130
   at Vintagestory.ClientNative.CrashReporter.Start(ThreadStart start) in VintagestoryLib\Client\ClientPlatform\ClientNative\CrashReporter.cs:line 93

**Comment by 46hawk** [comment]
Just ran into this one while jumping into water, not while dying. Attached my crash log, but it's looking pretty much the same as MtSheepFarm's
[client-crash.log](https://github.com/user-attachments/files/18141607/client-crash.log)

**Comment by radfast** [comment]
Thanks @MtSheepFarm @46hawk 

I agree, all those crash logs show the same issue, thank you it is helpful to confirm that.

OK this issue should, with any luck, be fixed in 1.20-rc.5 when it comes. No specific ETA, but I'd expect next few days.

Meanwhile, it's possible this crash is something which will just happen very occasionally, I hope not more than once per day. There's nothing you can do specifically to prevent or control it - except that I guess not dying during gameplay may help to reduce the chances. If it occurs again, re-join your game and continue playing.

(One mystery is why it affects you folks specifically, and not hundreds or thousands of other players. I believe the root cause is what coders call a 'race condition', things changing simultaneously on two different CPU threads, and it may be a chance result of your specific hardware, other people with fractionally different hardware have the same parts of the game processed at slightly different speeds. Ultimately, one of those computer-age mysteries...)

**Comment by Craluminum2413** [comment]
@radfast It happened again

**Comment by radfast** [comment]
Thanks, this issue is properly fixed in upcoming 1.20.0-rc.7  (hopefully coming out tomorrow, with a bit of luck)
